src/pages/Profile.tsx

```diff
@@ -109,6 +111,9 @@ const SubscriptionService = ({
   ) : null;
 };
 
+// max number of subscriptions loaded for each interaction (pagination)
+const subscriptionsLimit: number = 20;
```

Comment:
Just a style note: maybe we can user UPPER_SNAKE_CASE notation to emphatize it's a constant value
```suggestion
const SUBSCRIPTIONS_LIMIT = 20;
```
or maybe
```suggestion
const SUBSCRIPTIONS_PAGE_SIZE = 20;
```

(no need `: number` as it's inferred from the value `20`)

Reply:
SUBSCRIPTIONS_PAGE_SIZE sounds good